### PR TITLE
introduced the vote class

### PR DESCRIPTION
**Pull request by WouterPennings**
### DO NOT MERGE! 
just to leave feedback

### CODE REVIEW

File: src/vote.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+let vote = class {
```
Comment (WouterPennings) [inline src/vote.js:1]: The smart contract is called: "poll", this class is a interface to use that contract, so poll is a better class name

File: src/vote.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  constructor() {}
+
+  async BlockCount(provider) {
```
Comment (WouterPennings) [inline src/vote.js:4]: `BlockCount` should not be here. It is a function about the whole blockchain not our smart-contract

File: src/vote.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    console.log("Blockchain: " + (await provider.getBlockNumber()));
+  }
+
+  async PollCurrentStandings(pollSigner) {
```
Comment (WouterPennings) [inline src/vote.js:8]: All these functions should return the value not print them. You want the results to be displayed on the front-end.

Comment (WouterPennings) [reply src/vote.js:8]: Oh and (this is something that I did), but the functions: `currentStandings()` and `voteCount()` do not change a statevariable in the smart-contract and thus these do not need to be singed.

File: src/vote.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+let vote = class {
+  constructor() {}
```
Comment (WouterPennings) [inline src/vote.js:2]: In this class we want the poll contract, so it has to look something like this:
```js
constructor(contract_addres, provider) {
    this.poll = new ethers.Contract(contract_addres, abi, provider);
}
```

File: src/vote.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    let res = await pollSigner.currentStandings();
+    console.log("Current standings: " + res);
+  }
+
+  async PollVoteCount(pollSigner) {
+    let res = await pollSigner.voteCount();
+    console.log("Poll Vote Count: " + res);
+  }
+
+  async PollVote(pollSigner, vote) {
```
Comment (WouterPennings) [inline src/vote.js:18]: When calling this functions, the parameter should prob be the public key of a wallet and than sign it inside the function.

**Review by WouterPennings — COMMENTED** [review]
Nice start looks, the two main things are:

1. Seperation of the logic and "front-end" code.
2. returning of values instead of printing them.

If things are unclear or you need help, you can also call/message me to ask!

File: src/vote.js (hunk 1 above)
Comment (Simerblur) [reply src/vote.js:1]: "poll" is already used for ethers.Contract

**Pull request merged**